stdlib/buffer.mli

```diff
@@ -131,3 +131,23 @@ val add_channel : t -> in_channel -> int -> unit
 val output_buffer : out_channel -> t -> unit
 (** [output_buffer oc b] writes the current contents of buffer [b]
    on the output channel [oc]. *)
+
+
+(** {6 Iterators} *)
+
+val to_iter : t -> char Iter.t
+(** Iterate on the buffer, in increasing order
```

Comment:
Ping on that.  Also please make sure that all mutable data structures document the behavior of mutation during iteration (if only to say that the behavior is undefined).